test/sslapitest.c

```diff
@@ -6684,7 +6684,63 @@ static int test_ticket_callbacks(int tst)
  *                  sends CertificateRequest, client reads and ignores it
  * Test 6: TLSv1.3, server continues to read/write after client shutdown, client
  *                  doesn't read it
+ * Test 7: TLSv1.3, client does not shutdown properly,
+ *                  server does not set SSL_OP_IGNORE_UNEXPECTED_EOF,
+ *                  server should get SSL_ERROR_SSL
+ * Test 8: TLSv1.3, client does not shutdown properly,
+ *                  server sets SSL_OP_IGNORE_UNEXPECTED_EOF,
+ *                  server should get SSL_ERROR_ZERO_RETURN
  */
+
+static int test_incorrect_shutdown(int tst)
+{
+    SSL_CTX *cctx = NULL, *sctx = NULL;
+    SSL *clientssl = NULL, *serverssl = NULL;
+    int testresult = 0;
+    char buf[80];
+    SSL_SESSION *sess;
+    BIO *c2s;
+
+    if (!TEST_true(create_ssl_ctx_pair(libctx, TLS_server_method(),
+                                       TLS_client_method(), 0, 0,
+                                       &sctx, &cctx, cert, privkey)))
+        goto end;
+
+    if (tst == 8)
+        SSL_CTX_set_options(sctx, SSL_OP_IGNORE_UNEXPECTED_EOF);
+
+    if (!TEST_true(create_ssl_objects(sctx, cctx, &serverssl, &clientssl,
+                                            NULL, NULL)))
+        goto end;
+
+    if (!TEST_true(create_ssl_connection(serverssl, clientssl,
+                                              SSL_ERROR_NONE))
+            || !TEST_ptr_ne(sess = SSL_get_session(clientssl), NULL)
+            || !TEST_true(SSL_SESSION_is_resumable(sess)))
```

Comment:
The session things don't seem useful in this test

Reply:
Removed. Thanks!